src/set_theory/surreal/dyadic.lean

```diff
@@ -178,33 +178,56 @@ begin
     linarith },
 end
 
-/-- The map `dyadic_map` sends ⟦⟨m, 2^n⟩⟧ to m • half ^ n. -/
-def dyadic_map (x : localization (submonoid.powers (2 : ℤ))) : surreal :=
-localization.lift_on x (λ x y, x • pow_half (submonoid.log y)) $
-begin
-  intros m₁ m₂ n₁ n₂ h₁,
-  obtain ⟨⟨n₃, y₃, hn₃⟩, h₂⟩ := localization.r_iff_exists.mp h₁,
-  simp only [subtype.coe_mk, mul_eq_mul_right_iff] at h₂,
-  cases h₂,
-  { simp only,
-    obtain ⟨a₁, ha₁⟩ := n₁.prop,
-    obtain ⟨a₂, ha₂⟩ := n₂.prop,
-    have hn₁ : n₁ = submonoid.pow 2 a₁ := subtype.ext ha₁.symm,
-    have hn₂ : n₂ = submonoid.pow 2 a₂ := subtype.ext ha₂.symm,
+/-- The additive monoid morphism `dyadic_map` sends ⟦⟨m, 2^n⟩⟧ to m • half ^ n. -/
+def dyadic_map : localization.away (2 : ℤ) →+ surreal :=
+{ to_fun :=
+  λ x, localization.lift_on x (λ x y, x • pow_half (submonoid.log y)) $
+  begin
+    intros m₁ m₂ n₁ n₂ h₁,
+    obtain ⟨⟨n₃, y₃, hn₃⟩, h₂⟩ := localization.r_iff_exists.mp h₁,
+    simp only [subtype.coe_mk, mul_eq_mul_right_iff] at h₂,
+    cases h₂,
+    { simp only,
+      obtain ⟨a₁, ha₁⟩ := n₁.prop,
+      obtain ⟨a₂, ha₂⟩ := n₂.prop,
+      have hn₁ : n₁ = submonoid.pow 2 a₁ := subtype.ext ha₁.symm,
+      have hn₂ : n₂ = submonoid.pow 2 a₂ := subtype.ext ha₂.symm,
+      have h₂ : 1 < (2 : ℤ).nat_abs, from dec_trivial,
+      rw [hn₁, hn₂, submonoid.log_pow_int_eq_self h₂, submonoid.log_pow_int_eq_self h₂],
+      apply dyadic_aux,
+      rwa [ha₁, ha₂] },
+    { have := nat.one_le_pow y₃ 2 nat.succ_pos',
+      linarith }
+    end,
+  map_zero' :=
+  begin
+    rw ← localization.mk_zero 1,
+    apply localization.lift_on_mk,
+    apply_instance,
```

Comment:
we could just have `localization.lift_on_zero` right?

Reply:
This doesn't exist in the library. I can add it to ring.localization separately.

Reply:
this is what I was suggesting, yeah

Reply:
Cool, I'll do that.

Reply:
Does this look good?

```lean
lemma lift_on_zero {p : Type*} (f : ∀ (a : R) (b : M), p) (H) 
  : localization.lift_on 0 f H = f 0 1 :=
by rw [← localization.mk_zero 1, localization.lift_on_mk]
```

Reply:
@ericrbg The proof does type check. Just want to check that this would be the correct definition of `lift_on_zero`.

Reply:
lgtm:)

Reply:
Another option is this:

```lean
lemma lift_on_zero {p : Type*} (f : ∀ (a : R) (b : M), p) (H) (b : M) 
  : localization.lift_on 0 f H = f 0 b :=
by rw [← localization.mk_zero b, localization.lift_on_mk]
```

Reply:
i think the 1 version is better honestly